mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1644,6 +1645,17 @@ def _log_eval_table(self):
         data = data.assign(**columns)
         artifact_file_name = f"{metric_prefix}{_EVAL_TABLE_FILE_NAME}"
         mlflow.log_table(data, artifact_file=artifact_file_name)
+        if self.eval_results_path:
+            from pyspark.sql import SparkSession
+
+            spark_session = SparkSession.getActiveSession()
+            if not spark_session:
+                spark_session = _create_local_spark_session_for_evaluate()
```

Comment:
If PySpark isn't installed, then `spark_session` will still be `None`. I think `_create_local_spark_session_for_evaluate` should throw.